### ISSUE TITLE
SDDM silently hangs up when /var/lib is full

### ISSUE THREAD

**Issue by dvalter** [sddm/sddm#1120]
When there's no free disk space on /var/lib/sddm, sddm neither starts properly nor quits with helpful log message. It leaves black screen with cursor instead.

###  Steps to reproduce
1. Log out
1. Stop sddm (`systemctl stop sddm` for systemd distros)
1. Fill all the disk space at the /var/lib/sddm mountpoint (`dd if=/dev/zero of=/var/lib/zerofile bs=1M` if mountpoint is /var/lib) **WARNING! It may cause some damage to your system or/and services**
1. Start sddm (`systemctl start sddm` for systemd distros)
1. Observe the broken SDDM and non-informatimve logs (`journalctl -eu sddm` for systemd distros)
**Note, that if /var/log mountpoint is full either, systemd may leave your logs incomplete**

### Reproduced on
- Fedora 28 x86_64, _sddm-0.17.0-3.fc28.x86_64_
- KDE Neon (Ubuntu 18.04 amd64), _sddm 0.17.0+p18.04+git20181011.1524-0 amd64_

**Comment by plfiorini** [comment]
You cannot expect any service to check the status of your machine instead of you (the system administrator).

**Comment by ryanmusante** [comment]
When there is no free disk space, many other things will fail ahead of SDDM, no?

**Comment by dvalter** [comment]
> You cannot expect a service to check the status of your machine instead of you (the system administrator).

SDDM actually does it when it receives write error and fails to initialize properly. 
But it does not provide any information about the сause. 
Possibly sddm-helper could provide such an output. but due to #1081 it does not work

**Comment by dvalter** [comment]
> @dvalter When there is no free disk space, many other things will fail ahead of SDDM, no?

Not quite. SDDM runs as sddm, not root (at least in Neon and Fedora), so it means SDDM is not enabled to use reserved space. So system works mostly fine, but SDDM does not.

**Comment by plfiorini** [comment]
All we can do here is warn if the state file in `/var/lib/sddm` cannot be saved, reporting whatever generic error the operating system gives us but you might not see it if `/var/log` is full too.

**Comment by Pointedstick** [comment]
Similar issue with a non-interactive black screen + cursor seen in https://github.com/sddm/sddm/issues/1132 after setting a custom image background in the KDE Plasma SDDM KCM.
